rust/azure_iot_operations_connector/src/file_mount_azure_device_registry.rs

```diff
@@ -1,4 +1,1110 @@
+        file_mount_manager.add_device_endpoint(&device2_endpoint3, &device2_endpoint3_assets);
+
+        temp_env::with_vars(
+            [(
+                ADR_RESOURCES_NAME_MOUNT_PATH,
+                Some(file_mount_manager.path()),
+            )],
+            || {
+                let mount_path = get_mount_path().unwrap();
+
+                let device_endpoints = get_device_endpoint_names(mount_path.as_path()).unwrap();
+
+                assert_eq!(device_endpoints.len(), 3);
+                assert!(device_endpoints.contains(&device1_endpoint1));
+                assert!(device_endpoints.contains(&device1_endpoint2));
+                assert!(device_endpoints.contains(&device2_endpoint3));
+            },
+        );
+    }
+
+    #[test]
+    fn test_get_asset_names() {
+        let file_mount_manager = TempFileMountManager::new("test_mount");
+
+        let (device1_endpoint1, device1_endpoint1_assets) =
+            device_with_assets!("device1", "endpoint1", "asset1", "asset2", "asset3");
+
+        file_mount_manager.add_device_endpoint(&device1_endpoint1, &device1_endpoint1_assets);
+
+        let device1_endpoint1_assets_set: HashSet<AssetRef> =
+            HashSet::from_iter(device1_endpoint1_assets.clone());
+
+        temp_env::with_vars(
+            [(
+                ADR_RESOURCES_NAME_MOUNT_PATH,
+                Some(file_mount_manager.path()),
+            )],
+            || {
+                let mount_path = get_mount_path().unwrap();
+
+                let device_endpoints = get_device_endpoint_names(mount_path.as_path()).unwrap();
+
+                let mounted_device1_endpoint1_assets_set = HashSet::from_iter(
+                    get_asset_names(mount_path.as_path(), &device_endpoints[0]).unwrap(),
+                );
+
+                assert_eq!(
+                    mounted_device1_endpoint1_assets_set,
+                    device1_endpoint1_assets_set
+                );
+            },
+        );
+    }
+
+    #[tokio::test]
+    async fn test_device_endpoint_create_observation_pre_mounted_success() {
+        let file_mount_manager = TempFileMountManager::new("test_mount");
+
+        let (device1_endpoint1, device1_endpoint1_assets) =
+            device_with_assets!("device1", "endpoint1");
+        let (device1_endpoint2, device1_endpoint2_assets) =
+            device_with_assets!("device1", "endpoint2");
+        let (device2_endpoint3, device2_endpoint3_assets) =
+            device_with_assets!("device2", "endpoint3");
+
+        file_mount_manager.add_device_endpoint(&device1_endpoint1, &device1_endpoint1_assets);
+        file_mount_manager.add_device_endpoint(&device1_endpoint2, &device1_endpoint2_assets);
+        file_mount_manager.add_device_endpoint(&device2_endpoint3, &device2_endpoint3_assets);
+
+        let mut device_endpoints =
+            HashSet::from([device1_endpoint1, device1_endpoint2, device2_endpoint3]);
+
+        temp_env::async_with_vars(
```

Comment:
does this need an assert around it? Checking to make sure that an error in the async portion will get properly propagated to fail the test

Reply:
This just returns `()`, any failure inside will fail the test